core/includes/theme.inc

```diff
@@ -1458,7 +1458,7 @@ function theme_get_setting($setting_name, $theme = NULL) {
   // If the setting is not found in config, load the default from the theme's
   // info file.
   $value = $configs[$theme]->get($setting_name);
-  if (is_null($value)) {
+  if (is_null($value) && $theme != '') {
```

Comment:
Shouldn't we be checking for `$theme != ''` earlier, whenever `$theme` is being used as a key of the `$configs` array?

Reply:
yep!

Reply:
Agreed with @klonos, this added check seems to be too late: after `$theme` has already been used multiple times as an empty string. We should either add more checks above or find a way to consolidate them into a single `if` statement for when the global `$theme` is not set.

Reply:
So I think @jenlampton is on the right side here though: both D8 and D7 only check around that point for validity because it is assumed that `$theme` may be empty because `$theme = !empty($GLOBALS['theme_key']) ? $GLOBALS['theme_key'] : ''`. 
D8 uses `if (isset($themes[$theme])) {`. So I think for us: 
```
  if (is_null($value)) {
    $themes = list_themes();
    if (isset($themes[$theme])) {
```

Reply:
If I could get a 4th opinion on this, I could push a new PR. Anyone have thoughts? Its 2 v 2 so far, we need a tiebreaker. 😄

Reply:
Seems to me if after line 1450 there still is no theme, if `$theme` is empty string or NULL then we just exit early.

I understand what you're saying about d8 doing the check later but seems to me there's no point in going through all that if theme is blank.